lib/core/application/event/event_application_form_bloc/event_application_form_bloc.dart

```diff
@@ -0,0 +1,153 @@
+  EventApplicationFormBloc({this.initialQuestions})
+      : super(
+          EventApplicationFormBlocState(
+            fieldsState: {},
+            isValid: false,
+            answers: [],
+          ),
+        ) {
+    on<EventApplicationFormBlocEventInitFieldState>(initFieldState);
+    on<EventApplicationFormBlocEventUpdateField>(updateField);
+    on<EventApplicationFormBlocEventUpdateAnswer>(updateAnswer);
+  }
+
+  void initFieldState(
+    EventApplicationFormBlocEventInitFieldState event,
+    Emitter emit,
+  ) {
+    final user = event.user;
+    final applicationProfileFields =
+        event.event?.applicationProfileFields ?? [];
+    final applicationQuestions = event.event?.applicationQuestions ?? [];
+    final Map<String, String> initialFieldState = {};
+    final userMap = user?.toJson();
+    for (var applicationProfileField in applicationProfileFields) {
+      final key = applicationProfileField.field ?? '';
+      final value = userMap![StringUtils.snakeToCamel(key)];
+      initialFieldState[key] = value != null ? value.toString() : "";
+    }
+    emit(
+      state.copyWith(
+        fieldsState: initialFieldState,
+        answers: applicationQuestions
+            .map(
+              (item) => Input$EventApplicationAnswerInput(
+                answer: "",
+                question: item.id ?? '',
+              ),
+            )
+            .toList(),
+      ),
+    );
+  }
+
+  void updateField(
+    EventApplicationFormBlocEventUpdateField event,
+    Emitter emit,
+  ) {
+    final Map<String, String> newFieldState = {...state.fieldsState};
+    newFieldState[event.key!] = event.value!;
+    emit(
+      _validate(
+        state: state.copyWith(fieldsState: newFieldState),
+        event: event.event,
+      ),
+    );
+  }
+
+  void updateAnswer(
+    EventApplicationFormBlocEventUpdateAnswer event,
+    Emitter emit,
+  ) {
+    // final applicationQuestions = event.event?.applicationQuestions ?? [];
+    final newAnswers = state.answers.map((answer) {
+      if (answer.question == event.questionId) {
+        return answer.copyWith(answer: event.answer);
+      }
+      return answer;
+    }).toList();
+    emit(
+      _validate(
+        state: state.copyWith(answers: newAnswers),
+        event: event.event,
+      ),
+    );
+  }
+
+  EventApplicationFormBlocState _validate({
+    required EventApplicationFormBlocState state,
+    Event? event,
+  }) {
+    final applicationProfileFields = event?.applicationProfileFields ?? [];
+    final applicationQuestions = event?.applicationQuestions ?? [];
+    // Check profile required fields valid
+    final allProfileRequiredFields = state.fieldsState.entries.where((entry) {
+      final isRequiredField = applicationProfileFields.any(
+        (applicationProfileField) =>
+            applicationProfileField.field == entry.key &&
+            applicationProfileField.required == true,
+      );
+      return isRequiredField == true;
```

Comment:
return isRequired is enough.

Reply:
Why getting `allProfileRequiredFields` so complicated ? only need to filter out non-required `applicationProfileFields`